src/NetAppFiles/NetAppFiles/Pool/NewNetAppFilesPool.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.NetApp;
+using Microsoft.Azure.Management.NetApp.Models;
+
+namespace Microsoft.Azure.Commands.NetAppFiles.Pool
+{
+    [Cmdlet(
+        "New",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesPool",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesPool))]
+    [Alias("New-AnfPool")]
+    public class NewAzureRmNetAppFilesPool : AzureNetAppFilesCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The resource group of the ANF account")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The location of the resource")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.NetApp/netAppAccounts/capacityPools")]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccount",
+            nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        [Alias("PoolName")]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools",
+            nameof(ResourceGroupName),
+            nameof(AccountName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The size of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        public long? PoolSize { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The service level of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Standard", "Premium", "Extreme")]
+        public string ServiceLevel { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Tags")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            ParameterSetName = ParentObjectParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The account for the new pool object")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesAccount InputObject { get; set; }
```

Comment:
@leonardbf nit: please rename this to `AccountObject`

Reply:
Ok. Renamed.